bqskit/ir/gates/quditgate.py

```diff
@@ -0,0 +1,24 @@
+"""This module implements the QuditGate base class."""
+from __future__ import annotations
+
+from bqskit.ir.gate import Gate
+
+
+class QuditGate(Gate):
+    """A gate that acts on qudits."""
+
+    _num_levels: int
+
+    @property
+    def radixes(self) -> tuple[int, ...]:
+        """The number of orthogonal states for each qudit."""
+        return tuple([self.num_levels] * self.num_qudits)
```

Comment:
How should we handle hybrid-level gates? I understand that might be out-of-scope for now, but we discuss if even briefly for the future.

Reply:
By hybrid-level do you mean cases where we have qudits of different levels within the same circuit? In that case, we may need a list of num_levels, one for each qudit in the gate operation. We could also provide the user with the option of providing a list or a single number. In the second case it is assumed that all qudits have the same number of levels. What do you think?